### DISCUSSION TITLE
Empty response body for insert, even though code 201

### DISCUSSION THREAD

**Discussion by bkniffler** [supabase/supabase#4054] [Questions] [answered]
Calling a POST
I'm not getting any return data on inserts, even when using SERVICE_KEY. Here is a demo request:

```
curl --location --request POST 'https://wyggqsntrvmsbttsrmwx.supabase.co/rest/v1/items?columns=%22uniqueRef%22%2C%22description%22%2C%22data%22%2C%22type%22%2C%22connectionId%22' \
--header 'prefer: return=minimal' \
--header 'Accept: application/json' \
--header 'Content-Type: application/json' \
--header 'apikey: $SERVICE_KEY' \
--header 'authorization: Bearer $SERVICE_KEY' \
--data-raw '[
  {
    "uniqueRef": "4-0",
    "description": "Description 0",
    "data": { "id": "4-0", "propertyId": "0", "description": "Description 0" },
    "type": "reservation",
    "connectionId": "2a464bb7-367e-49d2-844a-cd969181993b"
  }
]'
```
$SERVICE_KEY is the value of my service key. I've also tried with normal authentication tokens.

The item is created in my DB, I'm getting response code 201. But the body is completely empty. Tried using return=minimal as visible above, but it still doesn't seem to work.

A GET request works fine and returns all the elements.

Any ideas what's wrong? Thanks :)

**Comment by soedirgo** [comment, marked as answer]
Hmm, can you try `return=representation`? `return=minimal` is for making the body empty, so might be the opposite of what you want.

**Reply by bkniffler** [reply]
You're right, thought minimal is `{ id }` only, but its indeed just blank. Thanks @soedirgo

**Reply by wedmonds** [reply]
This is still happening with rest v1 (recommended path from supabase).

**Reply by GaryAustin1** [reply]
> This is still happening with rest v1 (recommended path from supabase).

What is happening?  It is not supposed to return anything if you use return=minimal.

**Reply by wedmonds** [reply]
> > This is still happening with rest v1 (recommended path from supabase).
> 
> What is happening? It is not supposed to return anything if you use return=minimal.

Ah! I did not read that correctly -- I've been working on this for a few hours and my eyes must have crossed. I was thinking minimal meant ID only. Instead I used representation along with just selecting ID to get the desired return.
